spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/mongo/MongoClientFactory.java

```diff
@@ -39,102 +35,23 @@
  * @author Stephane Nicoll
  * @author Nasko Vasilev
  * @author Mark Paluch
+ * @author Christoph Strobl
  * @since 2.0.0
  */
-public class MongoClientFactory {
+public class MongoClientFactory extends MongoClientFactorySupport<MongoClient> {
```

Comment:
While I agree having a shared factory that behaves consistently, I am not sure we are ready to do this just yet. There's quite some history in the original MongoDB auto-configuration and sharing the code with the reactive counter part will change the semantics. I think we should keep the two separated for the time being and investigate what bits are problematic.